lib/ace/ext/papeeria/papeeria_latex_highlight_rules_test.js

```diff
@@ -8,108 +8,99 @@ define(function(require, exports, module) {
-            var result = tokenizer.getLineTokens("\\begin{equation} $ $$ \\] \\end{equation*} x^2 \\text{hey} \\alpha \\$ % this is a comment", "start");
-            var tokens = result.tokens;
-            var state = result.state;
-
-            assert.ok(state.length);
-            // the first 4 tokens are `\begin{equation}`
-            for (var i = 4; i < tokens.length; i++) {
-                assert(isType(tokens[i], EQUATION_TOKEN_TYPE));
-            }
-        },
+    var mathConstants = {}
+    mathConstants[MATH_ENVIRONMENT_DISPLAYED_NUMBERED_STATE] = {}
+    mathConstants[MATH_ENVIRONMENT_DISPLAYED_STATE] = {};
+    mathConstants[MATH_TEX_INLINE_STATE] = {};
+    mathConstants[MATH_TEX_DISPLAYED_STATE] = {};
+    mathConstants[MATH_LATEX_INLINE_STATE] = {};
+    mathConstants[MATH_LATEX_DISPLAYED_STATE] = {};
+
+    mathConstants[MATH_ENVIRONMENT_DISPLAYED_NUMBERED_STATE].start = "\\begin{equation}";
+    mathConstants[MATH_ENVIRONMENT_DISPLAYED_STATE].start = "\\begin{equation*}";
+    mathConstants[MATH_TEX_INLINE_STATE].start = "$";
+    mathConstants[MATH_TEX_DISPLAYED_STATE].start = "$$";
+    mathConstants[MATH_LATEX_INLINE_STATE].start = "\\(";
+    mathConstants[MATH_LATEX_DISPLAYED_STATE].start = "\\[";
+
+    mathConstants[MATH_ENVIRONMENT_DISPLAYED_NUMBERED_STATE].end = "\\end{equation}";
+    mathConstants[MATH_ENVIRONMENT_DISPLAYED_STATE].end = "\\end{equation*}";
+    mathConstants[MATH_TEX_INLINE_STATE].end = "$";
+    mathConstants[MATH_TEX_DISPLAYED_STATE].end = "$$";
+    mathConstants[MATH_LATEX_INLINE_STATE].end = "\\)";
+    mathConstants[MATH_LATEX_DISPLAYED_STATE].end = "\\]";
+
+    mathConstants[MATH_ENVIRONMENT_DISPLAYED_NUMBERED_STATE].length = 4;
+    mathConstants[MATH_ENVIRONMENT_DISPLAYED_STATE].length = 4;
+    mathConstants[MATH_TEX_INLINE_STATE].length = 1;
+    mathConstants[MATH_TEX_DISPLAYED_STATE].length = 1;
+    mathConstants[MATH_LATEX_INLINE_STATE].length = 1;
+    mathConstants[MATH_LATEX_DISPLAYED_STATE].length = 1;
 
-        "test: \"\\begin{equation*}\": basic and mismatching ends": function() {
+    module.exports = {
+        "test: basic testing": function() {
             var rules = new PapeeriaLatexHighlightRules().getRules();
             var tokenizer = new Tokenizer(rules);
-            var result = tokenizer.getLineTokens("\\begin{equation*} $ $$ \\] \\end{equation} x^2 \\text{hey} \\alpha \\$ % this is a comment", "start");
-            var tokens = result.tokens;
-            var state = result.state;
-
-            assert.ok(state.length);
-            // the first 4 tokens are `\begin{equation}`
-            for (var i = 4; i < tokens.length; i++) {
-                assert(isType(tokens[i], EQUATION_TOKEN_TYPE));
+            var basicString = " x^2 \\text{hey} \\alpha \\$ % this is a comment";
+            for (var state in mathConstants) {
+                var stateConstants = mathConstants[state];
+                var testString = stateConstants.start + basicString;
```

Comment:
Yeah, but wouldn't that be better, if we had separate tests for entering each equation state and then other tests for exiting it?